### DISCUSSION TITLE
Run Task of Project in Workspace?!

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [denoland/deno#25883] [Q&A] [answered]
So let's say I have multiple projects in a workspace with their respective deno.json files and tasks and so on.
How can I execute one of those tasks without changing to the directory first? Is there some workspace argument for deno run?

**Comment by marvinhagemeister** [comment]
Not supported at the moment. There is a a feature request for this here https://github.com/denoland/deno/issues/24991

**Comment by myty** [comment]
> So let's say I have multiple projects in a workspace with their respective deno.json files and tasks and so on. How can I execute one of those tasks without changing to the directory first? Is there some workspace argument for deno run?

https://jsr.io/@rsm-hcd/spino

We are exploring Deno workspaces, and this was an itch that wouldn't go away. It's very crude at this point, and it needs some cleanup, but handles the basic scenario of running all tasks/scripts that match a name or names.

**Comment by SebastienGllmt** [comment, marked as answer]
This is supported now. You can run a a task on a specific project using `deno task -f <project> <command>`

Here is the full output from `deno help task` that specifies all the relevant options

```
Options:
      --cwd <DIR>        Specify the directory to run the task in
  -f, --filter <filter>  Filter members of the workspace by name
  -r, --recursive        Run the task in all projects in the workspace
```

**Reply by blouflashdb** [reply]
Name is the name of the `name` property in deno.json and package.json?

**Reply by bartlomieju** [reply]
Correct

**Comment by derekdowling** [comment]
It would be great to document this here: https://docs.deno.com/runtime/fundamentals/workspaces/. I finally stumbled on this discussion thread after banging my head against the wall on this for the past few months.

**Comment by serujin** [comment]
For the next person coming here: 
- https://docs.deno.com/runtime/reference/cli/task/#workspace-support
